invisible_cities/io/mcinfo_io.py

```diff
@@ -275,88 +443,134 @@ def load_mcparticles_df(file_name: str) -> pd.DataFrame:
-                          'initial_momentum_z': p_tb.col('momentum')[:, 2],
-                          'kin_energy'        : p_tb.col('kin_energy'),
-                          'creator_proc'      : p_tb.col('creator_proc').astype('U20')})
-
-    # Adding event info
-    evt_part_df = extents[['last_particle', 'evt_number']]
-    evt_part_df.set_index('last_particle', inplace = True)
-    parts = parts.merge(evt_part_df         ,
-                        left_index  =   True,
-                        right_index =   True,
-                        how         = 'left')
-    parts.rename(columns={"evt_number": "event_id"}, inplace = True)
-    parts.event_id.fillna(method='bfill', inplace = True)
-    parts.event_id = parts.event_id.astype(int)
-
-    # Setting the indexes
-    parts.set_index(['event_id', 'particle_id'], inplace=True)
-
-    return parts
-
-
-def load_mcsensor_response(file_name: str,
-                           event_range=(0, int(1e9))) -> Mapping[int, Mapping[int, Waveform]]:
-
+    extents = load_dst(file_name, 'MC', 'extents')
     with tb.open_file(file_name, mode='r') as h5in:
-        return read_mcsns_response(h5in, event_range)
-
-
-def get_sensor_binning(file_name : str) -> Tuple:
+        p_tb = h5in.root.MC.particles
+
+        # Generating parts DataFrame
+        parts = pd.DataFrame({'particle_id'       : p_tb.col('particle_indx'),
+                              'particle_name'     : p_tb.col('particle_name').astype('U20'),
+                              'primary'           : p_tb.col('primary').astype('bool'),
+                              'mother_id'         : p_tb.col('mother_indx'),
+                              'initial_x'         : p_tb.col('initial_vertex')[:, 0],
+                              'initial_y'         : p_tb.col('initial_vertex')[:, 1],
+                              'initial_z'         : p_tb.col('initial_vertex')[:, 2],
+                              'initial_t'         : p_tb.col('initial_vertex')[:, 3],
+                              'final_x'           : p_tb.col('final_vertex')[:, 0],
+                              'final_y'           : p_tb.col('final_vertex')[:, 1],
+                              'final_z'           : p_tb.col('final_vertex')[:, 2],
+                              'final_t'           : p_tb.col('final_vertex')[:, 3],
+                              'initial_volume'    : p_tb.col('initial_volume').astype('U20'),
+                              'final_volume'      : p_tb.col('final_volume').astype('U20'),
+                              'initial_momentum_x': p_tb.col('momentum')[:, 0],
+                              'initial_momentum_y': p_tb.col('momentum')[:, 1],
+                              'initial_momentum_z': p_tb.col('momentum')[:, 2],
+                              'kin_energy'        : p_tb.col('kin_energy'),
+                              'creator_proc'      : p_tb.col('creator_proc').astype('U20')})
```

Comment:
Yes, confirmed. I think that we are ok with not mixing new and old format.

Reply:
If we want to make interface blind to the file format maybe would be a better idea to fill the missing columns with nans? Cause someone doing analysis with old and new files would still have to have 2 versions of the code to work with each type, and we wanted to avoid that?